File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {

        int p = 0;
        int q = 0;

        while(p<t.length() && q<s.length()){
            if(t.charAt(p)==s.charAt(q)) {
                p++;
                q++;
            }
            else{
                p++;
            }
        }


        if(q<s.length()) return false;

        return true;
        
    }
}

File: 392-is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        n, m = len(s), len(t)
        slow, fast = 0, 0

        while(slow<n and fast<m):
            if(s[slow] == t[fast]):
                slow += 1
            fast += 1

        return slow == n
        
            
        